### ISSUE TITLE
Crash when opening repo (submodule)

### ISSUE THREAD

**Issue by josharian** [laullon/gitx#202]
Opening a particular submodule consistently crashes gitx. (Opening other submodules fail, but without crashes. Haven't dug into it much yet.) Using the released v0.8.4 version.

Crash report at https://gist.github.com/3241672

(Updated to move long crash report to a gist.)

**Comment by joehoyle** [comment]
I also get this issue, error report here: 

https://gist.github.com/3181542

**Comment by protocool** [comment]
Also seeing problem on 0.8.4 version.

Looks like the layout of submodules has changed since upgrading to Xcode 4.4 and installing updated command line tools.

Now my submodules don't have a .git directory, they have a .git file, which references a 'gitdir' instead.

Perhaps this is the source of the crash because other projects which were cloned before my Xcode upgrade, and which have .git directories in submodules, do not cause the crash.

**Comment by josharian** [comment]
@protocool yep, I see the same thing, with git version `1.7.11.2`. That's gotta be it.

**Comment by protocool** [comment]
I just built from master and it looks like problem is corrected, it just hasn't made it into the released version of GitX yet.

**Comment by mschaaf** [comment]
@laullon Another reason for a new release.

**Comment by vitch** [comment]
I'm on `git version 1.7.11.3` and I'm having the same crash (with GitX v0.8.4). I updated to this fork because it started happening with the very old GitX I was using but it seems to be also happening on this fork. If it was due to a change in `git` itself that would explain why I'm suffering from this now and hadn't noticed it previously (I also updated `git` recently).

**Comment by vitch** [comment]
Building my own version with XCode 4.4.1 also fixed the problem for me. I'm not familiar with OSX development but I checkout out this repo and then opened GitX.xcodeproj in XCode and pressed run. I found the generated executable in `~/Library/Developer/Xcode/DerivedData/GitX-[some-ref]/Build/Products/Debug` and it doesn't crash when I open submodules. Adding the detail here incase someone else comes across this issue and is put off by the idea of compiling themselves.

**Comment by mattheworiordan** [comment]
I too have this issue, Gitx crashes when opening submodules.  Gitx version `0.8.4`, git version `1.7.11.1`, Mountain Lion `10.8.2`.  Crash log is here http://pastebin.com/1tZFguxN.  I've tried upgrading Git to `1.8.2` but unfortunately that made no difference.  Any ideas on how to use Gitx with submodules would be appreciated, workarounds welcome.

**Comment by josharian** [comment]
I switched to https://github.com/rowanj/gitx, and highly recommend it. And yes, it handles submodules well.

**Comment by mattheworiordan** [comment]
Thanks @josharian, that works well.
